amalgamation/amalgamation.py

```diff
@@ -31,7 +31,7 @@
     'nvml.h', 'opencv2/opencv.hpp', 'sys/stat.h', 'sys/types.h', 'cuda.h', 'cuda_fp16.h',
     'omp.h', 'execinfo.h', 'packet/sse-inl.h', 'emmintrin.h', 'thrust/device_vector.h',
     'cusolverDn.h', 'internal/concurrentqueue_internal_debug.h', 'relacy/relacy_std.hpp',
-    'relacy_shims.h'
+    'relacy_shims.h', 'ittnotify.h', 'shared_mutex'
```

Comment:
shared_mutex is only available in c++14 right? We only require c++11

Reply:
Optionally C++14 can be enabled in the CMakeLists for both dmlc-core and mxnet, in which case it will use shared_mutex -- currently in thread_group.h

Reply:
for example:
dmlccore_option(USE_CXX14_IF_AVAILABLE "Build with C++14 if the compiler supports it" OFF)